test/autoInject.js

```diff
@@ -4,6 +4,36 @@ var {expect} = require('chai');
 
 describe('autoInject', () => {
 
+    it('should not be subject to ReDoS', () => {
+        // This test will timeout if the bug is present.
+        var someComments = 'text/*'.repeat(1000000)
+        expect(() => async.autoInject({
+            someComments,
+            a () {}
+        })).to.throw()
+    });
+
+    it('should properly strip comments in argument definitions', () => {
+        var foo =
+            `(inline /* remove me */) => {return a}` +
+            `(
+                // remove this
+                singleline
+            ) => {return a}`
+        expect (() => async.autoInject({
+            ab (a) {
+                a = foo
+                return a;
+            }
+        })).to.deep.eql(
+            `(inline ) => {return a}` +
+            `(
+                
+                singleline
+            ) => {return a}`
+        )
+    });
```

Comment:
Here is a test that will fail if either halves of the regular expression are not working:

```suggestion
    it('should properly strip comments in argument definitions', (done) => {
        async.autoInject({
            task1: function(task2, /* ) */ callback) {
                callback(null, true);
            },
            task2: function task2(task3 // )
                ,callback) {
                    callback(null, true);
            },
            task3: function task3(callback) {
                callback(null, true);
            }
        },
        (err, result) => {
            expect(err).to.eql(null);
            expect(result).to.deep.eql({task1: true, task2: true, task3: true});
            done();
        });
    });
```

Reply:
Thank you for your input @Trott 
This looks so much better!